text/src/autogluon/text/automm/data/trivial_augmenter.py

```diff
@@ -0,0 +1,252 @@
+
+
+def _shear_y_impl(pil_img, level):
+    """Shear the image along the vertical axis with `level` magnitude."""
+    max = 0.3
+    level = scale_parameter(level, max, "float")
+    if random.random() > 0.5:
+        level = -level
+    return pil_img.transform(pil_img.size, Image.AFFINE, (1, 0, 0, level, 1, 0))
+
+
+shear_y = TransformT("ShearY", _shear_y_impl)
+
+
+def _translate_x_impl(pil_img, level):
+    """Translate the image in the horizontal direction by `level` number of pixels."""
+    max = 10
+    level = scale_parameter(level, max, "int")
+    if random.random() > 0.5:
+        level = -level
+    return pil_img.transform(pil_img.size, Image.AFFINE, (1, 0, level, 0, 1, 0))
+
+
+translate_x = TransformT("TranslateX", _translate_x_impl)
+
+
+def _translate_y_impl(pil_img, level):
+    """Translate the image in the vertical direction by `level` number of pixels."""
+    max = 10
+    level = scale_parameter(level, max, "int")
+    if random.random() > 0.5:
+        level = -level
+    return pil_img.transform(pil_img.size, Image.AFFINE, (1, 0, 0, 0, 1, level))
+
+
+translate_y = TransformT("TranslateY", _translate_y_impl)
+
+
+def set_image_augmentation_space(max_strength):
+    global image_parameter_max
+    image_parameter_max = max_strength
+    image_all_transform = [
+        identity,
+        auto_contrast,
+        equalize,
+        rotate,  # extra coin-flip
+        solarize,
+        color,  # enhancer
+        posterize,
+        contrast,  # enhancer
+        brightness,  # enhancer
+        sharpness,  # enhancer
+        shear_x,  # extra coin-flip
+        shear_y,  # extra coin-flip
+        translate_x,  # extra coin-flip
+        translate_y,  # extra coin-flip
+    ]
+    return image_all_transform
+
+
+def set_text_augmentation_space(max_strength):
+    assert max_strength <= 1.0, "Invalid maximum strength. Must <= 1.0"
+    global text_parameter_max
+    text_parameter_max = max_strength
+    text_all_transform = [
+        "identity",
+        "syn_replacement",
+        "random_delete",
+        "random_swap",
+        "insert_punc",
+    ]
+
+    try:
+        nltk.data.find("tagger/averaged_perceptron_tagger")
+    except LookupError:
+        nltk.download("averaged_perceptron_tagger")
+    try:
+        nltk.data.find("corpora/wordnet")
+    except LookupError:
+        nltk.download("wordnet")
+    try:
+        nltk.data.find("corpora/omw-1.4")
+    except LookupError:
+        nltk.download("omw-1.4")
+
+    return text_all_transform
+
+
+class TrivialAugment:
+    """
+    Implementation for TrivialAugment (https://arxiv.org/abs/2103.10158)
+    Random sample one operation from all_transform
+    Random a strength between [0, max_strength]
+    """
+
+    def __init__(self, datatype, max_strength) -> None:
+        assert max_strength > 0, "Invalid maximum strength. Must > 0"
+        self.max_strength = max_strength
+        self.data_type = datatype
+        if datatype == "img":
```

Comment:
Consider replace `img` and `text` with the `IMAGE` and `TEXT` from `automm/constants.py`